lib/perl/Genome/Config/AnalysisProject/SubjectMapping/Command/Predict/SomaticValidation.pm

```diff
@@ -0,0 +1,247 @@
+                    indel_variant_list_id => '',
+                    sv_variant_list_id => '',
+                    tag => 'tumor-only',
+                );
+                $self->add_subject_mapping(\%data);
+            }
+        }
+    }
+    my $writer = Genome::Utility::IO::SeparatedValueWriter->create(
+        output => $self->file_path,
+        separator => "\t",
+        headers => \@headers,
+        print_headers => 0,
+    );
+
+    for my $subject_mapping ($self->new_subject_mapping_set->members) {
+        $writer->write_one($subject_mapping);
+    }
+    $writer->output->close();
+
+    $self->status_message('Predicted '. $self->new_subject_mapping_set->size .' new subject mappings.');
+
+    return $self->new_subject_mapping_set->size;
+}
+
+
+sub existing_subject_mapping_set {
+    my $self = shift;
+
+    if ($self->_existing_subject_mapping_set) { return $self->_existing_subject_mapping_set; }
+
+    my $id_method = $self->sample_identifier;
+    my @subject_mappings;
+    for my $subject_mapping ($self->analysis_project->subject_mappings) {
+
+        my %data;
+        for my $subject_bridge ($subject_mapping->subject_bridges) {
+            $data{$subject_bridge->label} = $subject_bridge->subject->$id_method;
+        }
+
+        for my $input ($subject_mapping->inputs) {
+            $data{$input->key} = $input->value;
+        }
+
+        for my $tag ($subject_mapping->tags) {
+            if (defined($data{tag})) {
+                $self->fatal_message('Unable to handle multiple subject mapping tags!');
+            }
+            $data{tag} = $tag;
+        }
+        push @subject_mappings, \%data;
+    }
+
+    my $subject_mapping_set = Set::Scalar->new(@subject_mappings);
+    $self->_existing_subject_mapping_set($subject_mapping_set);
+
+    $self->status_message('Found '. $self->existing_subject_mapping_set->size .' existing subject mappings.');
+
+    return $self->_existing_subject_mapping_set;
+}
+
+sub new_subject_mapping_set {
+    my $self = shift;
+    if ($self->_new_subject_mapping_set) { return $self->_new_subject_mapping_set; }
+    my $new_subject_mapping_set = Set::Scalar->new();
+    $self->_new_subject_mapping_set($new_subject_mapping_set);
+    return $self->_new_subject_mapping_set;
+}
+
+sub add_subject_mapping {
+    my $self = shift;
+    my $data = shift;
+    if ($self->existing_subject_mapping_set->has($data)) {
+        $self->warning_message('Skipping existing subject mapping betweern tumor \''. $data->{tumor_sample} .'\' and normal \''. $data->{normal_sample} .'\'!');
+        return;
+    }
+    if ($self->new_subject_mapping_set->has($data)) {
+        $self->warning_message('New subject mapping already exists in set, skipping!');
+        return;
+    }
+    $self->new_subject_mapping_set->insert($data);
+    return 1;
+}
+
+sub resolve_samples_by_individual_id {
+    my $self = shift;
+
+    my @all_dna_samples = grep {$_->sample_type =~ /dna/i} $self->analysis_project->samples;
```

Comment:
This matches `cdna`, which is included in the results for `$sample->is_rna`.  Is that intended?